test/src/main/java/io/strimzi/test/BaseITST.java

```diff
@@ -0,0 +1,210 @@
+     */
+    protected void deleteCustomResources() {
+        Collections.reverse(deploymentResources);
+        for (String resource : deploymentResources) {
+            LOGGER.info("Deleting resources {}", resource);
+            KUBE_CLIENT.delete(resource);
+        }
+        deploymentResources.clear();
+    }
+
+    /**
+     * Delete custom resources such as templates. Resources are deleted in the reverse order than they were created.
+     */
+    protected void deleteCustomResources(String... resources) {
+        for (String resource : resources) {
+            LOGGER.info("Deleting resources {}", resource);
+            KUBE_CLIENT.delete(resource);
+            deploymentResources.remove(resource);
+        }
+    }
+
+    /**
+     * Prepare environment for cluster operator which includes creation of namespaces, custom resources and operator
+     * specific config files such as ServiceAccount, Roles and CRDs.
+     * @param clientNamespace namespace which will be created and used as default by kube client
+     * @param namespaces list of namespaces which will be created
+     * @param resources list of path to yaml files with resources specifications
+     */
+    protected void prepareEnvForOperator(String clientNamespace, List<String> namespaces, String... resources) {
+        createNamespaces(clientNamespace, namespaces);
+        createCustomResources(resources);
+        applyClusterOperatorInstallFiles();
+    }
+
+    /**
+     * Prepare environment for cluster operator which includes creation of namespaces, custom resources and operator
+     * specific config files such as ServiceAccount, Roles and CRDs.
+     * @param clientNamespace namespace which will be created and used as default by kube client
+     * @param resources list of path to yaml files with resources specifications
+     */
+    protected void prepareEnvForOperator(String clientNamespace, String... resources) {
+        prepareEnvForOperator(clientNamespace, Collections.singletonList(clientNamespace), resources);
+    }
+
+    /**
+     * Prepare environment for cluster operator which includes creation of namespaces, custom resources and operator
+     * specific config files such as ServiceAccount, Roles and CRDs.
+     * @param clientNamespace namespace which will be created and used as default by kube client
+     */
+    protected void prepareEnvForOperator(String clientNamespace) {
+        prepareEnvForOperator(clientNamespace, Collections.singletonList(clientNamespace));
+    }
+
+    /**
+     * Clear cluster from all created namespaces and configurations files for cluster operator.
+     */
+    protected void teardownEnvForOperator() {
+        deleteClusterOperatorInstallFiles();
+        deleteCustomResources();
+        deleteNamespaces();
+    }
+
+    @BeforeEach
+    void setTestName(TestInfo testInfo) {
+        testName = testInfo.getTestMethod().get().getName();
```

Comment:
I guess you can check NPE for Optional here and in method `createTestClassResources()`